### DISCUSSION TITLE
v0.2.0

### DISCUSSION THREAD

**Discussion by cristianberneanu** [diffix/desktop#141] [Announcements]
__Setup packages are NOT signed.__

### Version 0.2.0

- Added UI feedback on data export.
- Added support for column generalization.
- Removed anonymized count value in low count rows.
- Added relative noise column in combined view.
- Added side navigation in Notebooks.
- Sets a default path on data export.

<hr /><em>This discussion was created from the release <a href='https://github.com/diffix/publisher/releases/tag/v0.2.0'>v0.2.0</a>.</em>

**Comment by fjab** [comment]
I really, really like this 👍 

Couple comments:
 
In generalisation, it's unclear when it's enough to fill one field and when it isn't. 
- Filling "substring start" but not "substring length" leads to no generalisation, whereas 
- filling "substring length" but not "substring start" works with substring start = 1 as default (suggestion: default values could be shown in light grey if the box is not filled out).

Bug with substrings: if (substring length + substring start) > (length of string), it throws me an "Anonymization Error".

For the Anonymization Summary, it occurs to me that the distribution of distortions would be useful as well. Right now I see the average and the maximum distortion, but I can't really tell if there are just a few rows with really high distortion or how this comes together. Sorting by "Count noise" only helps if there are relatively few rows.

The above also shows: We need unified terminology here... what is it now, distortion or noise?

**Comment by cristianberneanu** [comment]
> The above also shows: We need unified terminology here... what is it now, distortion or noise?

In my mind, they are slightly different, noise makes me think of the SD of the generator and distortion makes me think of the final output value.

**Reply by fjab** [reply]
That may be the right way of thinking about it, but looking at our interface, that is absolutely not clear:

<img width="1040" alt="Screenshot 2021-09-02 at 09 31 32" src="https://user-images.githubusercontent.com/1407916/131801636-0019c3f2-bc2a-41ca-9f62-fa6cba892962.png">

"Maximum distortion" = highest "Count Noise"

**Reply by edongashi** [reply]
This appears to be inconsistent. We'll get back to it once we do the stats properly.

**Comment by cristianberneanu** [comment]
> Sorting by "Count noise" only helps if there are relatively few rows.

Sorting is not very relevant because we only show a subset of the output values (even after we'll make the summary accurate).

To improve sorting utility, we can either:

1. Select top/bottom X values for each column when sampling the output.
2. Offload sorting to the anonymization service to always get the actual top values for a column.

**Reply by fjab** [reply]
Good point.

I still maintain that we should just take a randomised sample (of 1000 rows or whatever) instead of the head. Then we don't have to do any other magic – top / bottom X rows may be the right approach for this specific problem, but likely not always.

**Comment by cristianberneanu** [comment]
What do you think about changing "Substring start / Substring length" to something shorter, like "Text start / Length"?

**Reply by fjab** [reply]
I have no problem with that. Although I also don't feel strongly either way; it never seemed too long to me.
